client/src/ui/components/Profile.tsx

```diff
@@ -0,0 +1,218 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faEdit } from "@fortawesome/free-solid-svg-icons";
+import { faStar } from "@fortawesome/free-solid-svg-icons";
+// import ThemeContext from "./ThemeContext";
+
+interface ProfileProps {
+  gameTags: string[];
+  username: string;
+  description: string;
+  profilePicture: string;
+  membership: boolean;
+  auth: boolean;
+}
+
+const Profile: React.FC<ProfileProps> = ({
+  profilePicture,
+  username,
+  description,
+  gameTags,
+  membership,
+  auth,
+}) => {
+  // const authenticated = useContext(Authenticated)
+  // Needs to pass a profile through
+  // let authenticated = true;
+
+  const [editDescription, setEditDescription] = useState(description);
+  const [tempDescription, setTempDescription] = useState(description);
+  const [editProfilePic, setEditProfilePic] = useState(profilePicture);
+  const [tempProfilePic, setTempProfilePic] = useState(profilePicture);
+  const [editGameTags, setEditGameTags] = useState(gameTags);
+  const [tempGameTags, setTempGameTags] = useState(gameTags);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const [selectedOption, setSelectedOption] = useState<string[]>(gameTags);
+
+  const imgInput = useRef<HTMLInputElement>(null);
+
+  const handleSelectionChange = (selectedValues: string[]) => {
+    console.log("New dropdown selection: ", [...new Set(selectedValues)]);
+    setSelectedOption([...new Set(selectedValues)]);
+  };
+
+  const colorClasses = [
+    "text-red",
+    "text-blue",
+    "text-green",
+    "text-yellow",
+    "text-pink",
+    "text-orange",
+  ];
+
+  function handleEditClick() {
+    setIsEditing(!isEditing);
+    setTempDescription(editDescription);
+    setTempProfilePic(editProfilePic);
+    setTempGameTags(editGameTags);
+  }
+
+  function handleSaveClick() {
+    // Sends updates to backend
+    setEditDescription(tempDescription);
+    setEditGameTags(selectedOption);
+    setEditProfilePic(tempProfilePic);
+    setIsEditing(!isEditing);
+  }
+
+  function handlePFPClick() {
+    // Updates the profile picture only when editing mode is on
+    if (isEditing) {
+      console.log("PFP is being edited");
+      imgInput.current?.click();
+    }
+  }
+
+  const handleProfilePicChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    // Assert event.target as HTMLInputElement
+    const input = event.target as HTMLInputElement;
+
+    // Check if files exist and if the first file is defined
+    if (input.files && input.files[0]) {
+      const file = input.files[0];
+
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        // Make sure to check if reader.result is a string before setting state
+        if (typeof reader.result === "string") {
+          setTempProfilePic(reader.result);
+        }
+      };
+      reader.readAsDataURL(file);
+    } else {
+      console.log("No file chosen or file access error.");
+    }
+  };
+
+  function list() {
```

Comment:
Two things
1. list() is a pretty generic name, I would change it to `getGameList()` or something similar (although we might remove this function as per the below point)

2. I assume this function is meant to be a temp placeholder and eventually we'll want to change this to an async function to the backend to get the actual list? In that case, I'd just save this as `gameList` in this component's state with `useState` and have it default to this "master game list", then when we connect the database later, we can just add in a `useEffect` to set the game list state